libnmstate/nm/dns.py

```diff
@@ -134,3 +135,44 @@ def find_interfaces_for_name_servers(iface_routes):
     """
     return (nm_route.get_static_gateway_iface(Interface.IPV4, iface_routes),
             nm_route.get_static_gateway_iface(Interface.IPV6, iface_routes))
+
+
+def get_iface_dns_metadata(acs_and_ipv4_profiles, acs_and_ipv6_profiles):
```

Comment:
This is not metadata but dns config. The caller may use the data retrieved as metadata of an interface.